packages/core/src/Stores/Modules/Cashier/cashier-store.js

```diff
@@ -266,6 +289,14 @@ export default class CashierStore extends BaseStore {
         }
     }
 
+    @action.bound
+    setIsP2PBannerVisible(is_visible) {
+        this.is_p2p_visible = is_visible;
+        if (!is_visible) {
+            this.is_p2p_visible = false;
```

Comment:
I'm confused, this is the same flag as `setIsP2pVisible` function, and why is the if condition needed? line 294 already sets it to false if is_visible is passed as false

Reply:
if we need a separate function so it doesn't reroute to deposit, we should keep the common functionality in one function and the routing in a separate function so we don't duplicate the same code. we should only set `is_p2p_visible` in one place.

Reply:
Oh my mistake, I mistaken the functionality of `setIsP2pVisible`. It's perfectly fine in this use case and will be sticking to  that.